File: WordsDataFrame.java
import java.util.ArrayList;

public class WordsDataFrame extends DataFrame {
    public WordsDataFrame(String[][] data, String[] columns) {
        super(data, columns);
    }

    public static WordsDataFrame build(NormalDataFrame nndf, MessageTypeDataFrame mtdf, boolean run) {
        if (!run) {
            return null;
        }
        ArrayList<String> words = new ArrayList<>();
        ArrayList<Integer> rarities = new ArrayList<>();
        NormalDataFrame ndf = new NormalDataFrame(nndf);
        ndf.filter(mtdf.select("index").toArray());

        for (int i : ndf.indexing) {
            String text = ndf.row(i)[3];
            for (String word : text.split(" ")) {
                if (word.equals("")) // This will occur if 'text' has internal adjacent double whitespaces.
                    continue;
                if (!words.contains(word)) {
                    words.add(word);
                    rarities.add(0);
                }
                int index = words.indexOf(word);
                rarities.set(index, rarities.get(index) + Util.count(text, word));
            }
        }
        WordsDataFrame toReturn = new WordsDataFrame(new String[][] {Util.toStringArray(words), Util.toStringArray(rarities)},
                new String[]{"word", "rarity"});
        toReturn.sortValues("rarity");
        toReturn.reverse();
        return toReturn;
    }
}

File: WordsDataFrame.py
# Words DF -------------------------------------------------------------------------------------------------------------
import pandas as pd
from timeit import default_timer as timer

import NormalDataFrame
import util


def build(df, mtdf, run):
    if not run:
        util.append_time(0)
        return None
    temp = timer()
    df = df.drop(mtdf['ind'])
    words = []
    rarities = []

    for i in list(df.index):
        date, time, author, text = NormalDataFrame.row(df, i)
        if util.is_nan(text):
            continue
        for word in text.split():
            if word not in words:
                words.append(word)
                rarities.append(0)
        for word in text.split():
            rarities[words.index(word)] += text.count(word)

    util.append_time(timer() - temp)
    return pd.DataFrame(list(zip(words, rarities)), columns=['word', 'rarity']).sort_values(by="rarity",
                                                                                            ascending=False)


def row(wdf, i):
    return (wdf.at[i, "word"],
            wdf.at[i, "rarity"])
